otri/filtering/filters/nuplicator_filter.py

```diff
@@ -0,0 +1,45 @@
+from ..filter import Filter, Stream, Collection
+import copy
+
+
+class NUplicatorFilter(Filter):
+    '''
+    N-uplicates the input stream. Placing a copy of the input in each output filter.
+    Inputs: a single Stream.
+    Outputs: Any number of streams.
+    '''
+
+    def __init__(self, source_stream: Stream, output_streams_count: int, deep_copy: bool = False):
+        '''
+        Parameters:
+            source_stream : Stream
+                A single Stream that must be n-uplicated
+            output_streams_count : int
+                The number of output streams for this filter
+            deep_copy : bool = False
```

Comment:
I'd set it to True by default, I can't think of a program that would make use of n-upling pointers to the same object